### ISSUE TITLE
Provider "allanime": Specify dub over sub?

### ISSUE THREAD

**Issue by IngwiePhoenix** [justfoolingaround/animdl#206]
**The inquiry**

Long story short: How do I specify to download a dubbed version instead of the subbed one? The README didn't seem to feature a sub/dub selector.

Thanks! ^^

**Comment by toomyzoom** [comment]
Do you have the series name in question with dub? When I tried spy x family season 1 which I checked on the the site, animdl only returns sub, no dub included.
If the dub is included, you may use the quality switch with regular expression to filter selection but I don't think it applies to this case. @justfoolingaround will need to update the parser for `allanime` to include the dub.
The quality switch can be found in the manual at section:
[https://github.com/justfoolingaround/animdl#-q----quality-argument](url)

**Comment by IngwiePhoenix** [comment]
Yeah, it has a dub: https://allanime.site/anime/XWmFqfG6RQgagevEn

When I did a `animdl grab` for it, it only returned one stream. In fact, here is a snippet:

```json
 "vrv": {"stream_type": "Hls", "provider_configuration": {"subtitle_language": "en-US"}}}]}' | jq
{
  "episode": 13,
  "streams": [
    {
      "stream_url": "https://video.wixstatic.com/video/e3082c_fb8cc77280184013a0920313d13c15e9/480p/mp4/file.mp4",
      "quality": 480,
      "vrv": {
        "stream_type": "Hls",
        "provider_configuration": {
          "subtitle_language": "en-US"
        }
      }
    }
  ]
}
```

Maybe the grabber can't see the dubs at its current state?

**Comment by justfoolingaround** [comment]
You can make AllAnime (only this provider) select dub by using configurations. In essence,

```yml
superanime:
  type_of: "sub"
```

**Comment by IngwiePhoenix** [comment]
Can I see all the possible configuration options somewhere? Might find a few more that I could adjust.

Thank you for the hint for this one, will try it out later to see if it works!

**Comment by xxBreadGodxx** [comment]
Was this ever figured out? I downloaded the source went into animdl->core->config->__init__.py I found the default settings there where it tries to find the best one with subtitles. I tried changing it to "dub" or "dubbed" since the the quality string just has it set to "best[subtitle]/best"

**Comment by GauravMohla** [comment]
@xxBreadGodxx the following works for me:

`superanime:
    type_of: "dub"`

created a file in userprofle and it works fine

**Comment by TZR-Redstonemaniac** [comment]
> @GauravMohla I ended up reconfiguring the source. I appreciate it though, windows powershell is really picky with this and wouldn't let me do options regardless so I ended up setting it to dub, changing a few lines and getting rid of the line that grabs the subtitles for the dub and if it doesn't find it in dub it auto grabs sub instead. But hey if anyone wants to have subtitles with their dub let me know, its easy to set up in source.

How did you do it?
